src/utils/typedefs-wrapper.cpp

```diff
@@ -44,26 +46,165 @@ namespace detail {
     }
     Il2CppString* alloc_str(std::u16string_view str) {
         il2cpp_functions::Init();
-        return il2cpp_functions::string_new_utf16((Il2CppChar*) str.data(), str.size());
+        return il2cpp_functions::string_new_utf16((Il2CppChar const*) str.data(), str.size());
+    }
+
+    Il2CppString* strappend(Il2CppString const* lhs, Il2CppString const* rhs) noexcept {
+        if (!lhs && !rhs) return il2cpp_functions::string_new_utf16(u"", 0);
```

Comment:
do you think it would make more sense to just ret null here? I think it might. Curious what you think though.

Reply:
hmmm, my thought here was that 2 nullptr strings would be an empty string if appended, but nullptr makes sense too